src/google/cloud/happybase/table.py

```diff
@@ -143,19 +143,23 @@ def families(self):
     def regions(self):
         """Retrieve the regions for this table.
 
-        .. warning::
-
-            Cloud Bigtable does not give information about how a table is laid
-            out in memory, so this method does not work. It is
-            provided simply for compatibility.
-
-        :raises: :class:`NotImplementedError <exceptions.NotImplementedError>`
-                 always
+        :rtype: list
+        :returns: A list of :class:`~google.cloud.bigtable.happybase.region_locator.RegionLocation` objects.
+                  lists regions for every row key of sample row keys of this table. every region's start key is previous row
+                  key if it differs from current row key or None byte and end key is current row key. if table have not any
+                  sample row key it return list with one region in with None byte start key and None byte end key.
         """
-        raise NotImplementedError(
-            "The Cloud Bigtable API does not have a "
-            "concept of splitting a table into regions."
-        )
+        regions = []
+        start_key = b""
+        for sample_row_key in self._low_level_table.sample_row_keys():
+            end_key = sample_row_key.row_key
+            if start_key is not end_key:
+                regions.append(RegionLocation(start_key, end_key))
```

Comment:
Please use a `dict` here instead of `RegionLocation`. There's no need for `RegionLocation` at all, so please remove it as well.

Reply:
removed